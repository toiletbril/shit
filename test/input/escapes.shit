echo \*\*;
echo \~\*\*;
echo \\\\;
echo data/*/\*;
test -e ~/\* && no such file;
test -d \~ && echo no such dir;
echo a'''\ hello'\ \ world"lol" "\"";
echo "*"
echo '*'
