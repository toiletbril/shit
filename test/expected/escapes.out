[CompoundList]
  [CompoundListCondition, None]
    [SimpleCommand "echo **"]
  [CompoundListCondition, None]
    [SimpleCommand "echo ~**"]
  [CompoundListCondition, None]
    [SimpleCommand "echo \\"]
  [CompoundListCondition, None]
    [SimpleCommand "echo data/*/*"]
  [CompoundListCondition, None]
    [SimpleCommand "test -e ~/*"]
  [CompoundListCondition, &&]
    [SimpleCommand "no such file"]
  [CompoundListCondition, None]
    [SimpleCommand "test -d ~"]
  [CompoundListCondition, &&]
    [SimpleCommand "echo no such dir"]
  [CompoundListCondition, None]
    [SimpleCommand "echo a\ hello  worldlol ""]
  [CompoundListCondition, None]
    [SimpleCommand "echo *"]
  [CompoundListCondition, None]
    [SimpleCommand "echo *"]
[Escape Map
  00000110 00000000 11100000 00000110 00000000 00000001 00000000 00000100 00000000 00000000 00000000 01000000 00000000 00000000 00000001 00000011 00000000 00000000 01000000 00001000 00000100 
]
**
~**
\\
data/abcd/* data/abcdefg/*
a\ hello  worldlol "
*
*
[Code 0]
